Core/src/Detector/LayerStructureBuilder.cpp

```diff
@@ -49,17 +49,20 @@ namespace {
 ///
 /// @return a configured surface candidate updators
 template <Acts::detail::AxisBoundaryType aType>
-Acts::Experimental::SurfaceCandidatesUpdator createUpdator(
+Acts::Experimental::SurfaceCandidatesDelegate createUpdator(
     const Acts::GeometryContext& gctx,
-    const std::vector<std::shared_ptr<Acts::Surface>>& lSurfaces,
-    const std::vector<size_t>& assignToAll,
+    std::vector<std::shared_ptr<Acts::Surface>> lSurfaces,
+    std::vector<size_t> assignToAll,
```

Comment:
Just need to make sure we don't copy these needlessly.

Reply:
agreed - I just think by value gives you more flexibility when you are sinking the param anyways